### DISCUSSION TITLE
Downloaded video doesn't have any graphics

### DISCUSSION THREAD

**Discussion by mredowan** [aclap-dev/video-downloadhelper#761] [Help and Q&A]
### Your question

Hi
I downloaded some Udemy course videos on my local drive/desktop, but the videos don't show anything when playing. Only thousands of boxes blink on the screen!
Could you be kind enough to suggest to me what to do now?
![Screenshot](https://github.com/aclap-dev/video-downloadhelper/assets/131864336/e7c343bb-9b59-4bf5-82bb-c989792a2c0b)

### Addon & CoApp version and details

version: 9.0.1.22
target: google
channel: stable
lang: en-GB
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"accepted":true,"email":"<removed>","key":"bc6c5d8b7d1b46a8"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36
{
    "valid_license_message_has_been_displayed": true,
    "used_sidebar_button": true,
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "use_sidebar": true,
    "open_count_store": 4,
    "successfull_dl": 3
}

**Comment by Mattmjs** [comment]
Nothing, Udemy is protecting the content with DRM.

**Reply by mredowan** [reply]
but I downloaded many other videos from the same DRM-protected Udemy courses using this app! How did that happen?

**Reply by reginafer12** [reply]
It is happening the same to me.

**Comment by Smietje77** [comment]
Same problem here; I downloaded the first 6 episodes with no incidents, and then boom...

**Comment by GrampaWildWilly** [comment]
What the original poster showed in his post to start this discussion is a textbook example of what a video looks like when it's protected by DRM but you download it anyway.  Many people have posted in this forum & in our old forum back on Google that both Hotmart & Udemy have problems like this.  One fellow user posted that it seemed like lesson 1 of a course would be downloadable by VDH but the second & subsequent lessons in the course would be protected by DRM.  DRM seems to have been a recent change in policy on these sites.  You might find older content that is not protected by DRM & newer content that is.  I'm not a member of either site so I can't check this, but I suspect the various authors decide independently whether to apply DRM protection.  That means you'll find some content VDH can download without problems & some content that will give results like in the original post above.  Do not expect VDH to ever provide a solution.  VDH has never, does not now, & will never download content protected by DRM  The VDH guys (I'm just another user, not an employee of VDH) have repeatedly posted exactly that comment.  So just give up.  Maybe you can get a screen recorder to record your courses.  Don't hold out too much hope on that score, either.  It seems that only expensive apps associated with expensive services will do screen recording these days.

This is the sad reality of things.

**Reply by rihntintin** [reply]
Hi [https://github.com/mredowan](url)
I'm just a VDH user, not a VDH support or developer, so I don't know a lot, but here goes.
I've used VDH since 2008 and Udemy since 2004. I had issues with downloading some, just some, of Udemy course, but not all of them, and even within the same course, I was able to successfully download the first 10 lectures, but none of the rest of the lectures. It is a mixed bag of successful and unsuccessful downloads from Udemy. It's not consistent with all Udemy course and therefore difficult to understand what makes some courses and/or some lectures fine and others unusable. Plus, it seems that all Preview lectures are all fine. But what has helped me a lot was buying a new iMac. Many of the issues went away and some of issues still persist, and I don't know why. Try using a friends PC or MacOS and see if it helps.

If you want to list some of the Udemy courses you have issues with, I will see if I purchased them and I'll try it myself. Most of my courses are for Python, some for DevOps, ChatGPT, etc. and other IT related courses and some non-IT courses. But maybe your interests lie elsewhere. 

Anyway, this is for you and for the VDH developers and I will create my question later this week when I have more free time. Around March 22, 2024, VDH would display if the lectures were HLS Streaming or DASH Streaming. VDH doesn't do this anymore with the latest release in May. The HLS lectures were successful in downloading all the time, but the DASH lectures never were successful, and there were never any error messages. The DASH lectures created files similar to your screenshot, but most of downloads displayed a solid colored screen and as it was playing, the entire screen would flash a new color for a split second and go back to the original color. It would repeat this flashing with different colors, while it was playing, but there was never any audio or video displayed.

This is the course I was referring to: [https://www.udemy.com/course/the-data-analyst-course-complete-data-analyst-bootcamp/](url)

I hope this is useful to anyone. Here is my system information:

version: 9.0.2.3
target: google
channel: stable
lang: en-US
coapp: {"found":true,"new_version":false,"path":"/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp","version":"2.0.19"}
license: {"accepted":true,"email":"<removed>","key":"1ced0282b1ab4aff"}
platform: x86-64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36
{
    "valid_license_message_has_been_displayed": true,
    "used_sidebar_button": true,
    "used_popup_button": true,
    "download_directory": "/Users/bonnieoliverimac2017/Movies/Udemy",
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "use_sidebar": false,
    "open_count_store": 34,
    "successfull_dl": 14
}

**Reply by rihntintin** [reply]
I just noticed the successful HLS files have a Codecs of AAC, H.264 & the DASH lectures have MPEG-4 HE AAC, H.264.
I don't know what that means nor if it is useful.

Plus, you should download the VLC free program. It plays mp4 files and it has some nice options compared to QuickTime Player.

**Reply by rihntintin** [reply]
Never mind. You're already using VLC.

**Reply by GrampaWildWilly** [reply]
Some speculation here.  VDH displays what it finds.  Maybe some content used to be either HLS or DASH, & now it's something else.  You never know.  They could reorganize their web site on a whim at any time.  In place of HLS or DASH, other things I've seen there are RawHLS, HTTPMedia, and maybe 2 or 3 others I'm not recalling at the moment.  What things are you seeing instead of either HLS or DASH?

I think VDH is having a problem at the moment with DASH content.  I believe @paulrouget Paul is working on it.  By the way, that's how you tag another user: @ followed by the user name.  Type @ in your post & see what it gives you.  Just @.  Nothing more.

Your description of the flashing solid color screen is, like I said above, textbook proof of DRM.  Maybe it's green, maybe it's black, maybe it's some other color.  The color isn't particularly significant.  The unusable video is the proof it was protected by DRM.  You are confirming what I have long suspected, that the application of DRM protection appears to be at the whim of the author.  There is probably no explaining it.  It's probably a bit capricious.

If you are so motivated, I have a Plan B.  Of course, I can't give you examples on Udemy (or Hotmart) since I'm not a member.  But you might be so motivated.  You might search this forum for threads in which I implement my Plan B.  I've tried to be consistent about saying Plan B in those threads so that they could be found by searching this forum with "plan b" as your search key.  Plan B is just an attempt to download the content using ffmpeg directly instead of VDH.  Everybody here has ffmpeg.  It's part of the CoApp.  Whatever directory you installed the CoApp in also contains ffmpeg.  This is a version that is somewhat enhanced over what you can get from ffmpeg.org.  In the old days, VDH did not use ffmpeg to download content.  But since about a year ago & Paul's addition to the dev team, the CoApp has been using ffmpeg to download streams.  When I try to apply Plan B, the ffmpeg I use is the one that comes with the CoApp.  Because of that, when I get Plan B to work, it indicates that there is a good chance Paul can make VDH work on that content as well . . . eventually.  Good chance.  No guarantees.  But I believe the odds are good.

In addition to those examples, I wrote a tutorial on how to use ffmpeg back on our old Google forum.  If you enter the search key "vdh ffmpeg" into Google, the first search result is the link to the tutorial.  I can't say what results you might get from Yahoo, Bing, Duckduckgo, or any other search engine.  But Google gives you that result.  For content that is NOT protected by DRM, you might try your hand at applying Plan B yourself.  You're a member so there's a good chance you can make it work, which would be a great help to your fellow Udemy members.  Plan B fails just like VDH does on content protected by DRM.  It downloads something but you'll see weird messages in the ffmpeg log & the resulting "video" will be the corrupted garbage you've described.

**Reply by GrampaWildWilly** [reply]
Yes.  VLC has always been the recommended media player around here.

The codec stuff is kind of incidentally interesting.  Both of the combinations you cite are indicative of MP4 content.  There are other combinations you'll see on MKV content.  Those things just are what they are.  They are not predictive of success or failure of VDH.

**Reply by Mattmjs** [reply]
About the old google forum, I noticed in the past day or so it got banned again. So it was a good decision to move to Github.

**Reply by GrampaWildWilly** [reply]
Such a pain in the butt.  I guess we were banned for inactivity.  It's been at least a few weeks that posting there has been blocked.  So it has to be due to some frivolous accusation by someone with an axe to grind.  Despite that, there was still some useful content on there if anybody happened to go mine it.  I suppose Michel isn't going to bother fixing it this time.  It took a threat of a lawsuit last time to get it back online.  But the expectation that this would happen again was the primary reason we did move here.  Good riddance, I suppose.

So much for my ffmpeg tutorial.  And our brilliant discussion of the evasive maneuvers we concocted to get things when VDH couldn't.  And my tutorial on the alternative method of downloading from YouTube.  And maybe a dozen other probably still useful discussions.  Oh well.  And so it goes.

**Reply by GrampaWildWilly** [reply]
The Google search on "vdh ffmpeg" does still give the link to the tutorial.  But when you click the link, you get the canned response about the group being banned.  Wankers.
